soc/realtek/ec/rts5912/reg/reg_crypto.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2025 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_REALTEK_RTS5912_REG_CRYPTO_H
+#define ZEPHYR_SOC_REALTEK_RTS5912_REG_CRYPTO_H
+
+struct sha2_type {
+	uint32_t ctrl;
+	const uint32_t status;
+	const uint32_t reserved[2];
+	uint32_t digest[16];
+	uint32_t fifo_in;
+};
```

Comment:
The `__packed` attribute should be added to the structure `struct sha2_type` definition not to the users of this structure.
```suggestion
} __packed;
```

Reply:
Thanks for pointing this out @keith-zephyr. This was the structure I was meant to point to in [my comment](https://github.com/zephyrproject-rtos/zephyr/pull/89263#discussion_r2095465275), but for some (unknown) reason then I pointed to the wrong ones. Sorry @JasonLin-RealTek for the confusion.

Reply:
We can't use __packed here. The system will shutdown and turn on it over and over again.

Reply:
That is strange.  I downloaded your patch, and added the following checks:

```
BUILD_ASSERT(sizeof(struct sha2_type) == 84, "ERROR - invalid size for SHA2 registers");
BUILD_ASSERT(sizeof(struct sha2dma_type) == 932, "ERROR - invalid size for SHA2DMA registers");
```

As a test, I enabled `CONFIG_CRYPTO` on the rts5812_evb board locally.  The checks above pass with and without the `__packed` attribute set when using  the Zephyr toolchain.   The crash may be due to other code changes, or possibly a compiler bug.

Because all the members are 32-bit, the `__packed` attribute shouldn't do anything, so this is not a blocker for me.

Reply:
> We can't use __packed here. The system will shutdown and turn on it over and over again.

This sounds like an hard fault to me and it's something that should be investigated. As @keith-zephyr correctly pointed out in his last comment (thanks for the build tests!), adding `__packed` shouldn't change anything because the CPU core is 32 bits and all the registers are 32 bit, so the structure should have the same "compactness" no matter if `__packed` is used or not.
I can accept this change as well, but I strongly encourage you to run some test locally to understand what happens when `__packed` is added. It might not affect you as developer of this driver, but it might affect any end user trying to do the same on that platform.

Reply:
The `__packed` is definitely not necessary here, there is no padding needed in this structure and it is 4 bytes aligned as well. Does it a require a different alignment (though the packing should not affect it) ?

It looks to me that there is something other that this attribute is masking.

Reply:
We won't add __packed on this structure now.